Aztec/Classes/Public/TextKit/TextAttachment.swift

```diff
@@ -116,6 +129,43 @@ public class TextAttachment: NSTextAttachment
 
         return CGRect(origin: CGPointZero, size: CGSize(width: width, height: onScreenHeight(width)))
     }
+
+    func updateImage() {
+        switch kind {
+        case .RemoteImage:
+            break
+        default:
+            return
+        }
+        guard !isFetchingImage else {
+            return
+        }
+        isFetchingImage = true
+
+        let imageURL: NSURL
+        switch kind {
+        case .RemoteImage(let url):
+            imageURL = url
+        default:
+            return
+        }
+
+        image = imageProvider?.image(forURL: imageURL, inAttachment: self,
+                                                onSuccess: { [weak self](image) in
+                                                    self?.isFetchingImage = false
+                                                    self?.image = image
+                                                    self?.kind = .RemoteImageDownloaded(url: imageURL, image: image)
+                                                    self?.triggerUpdate()
+            }, onFailure: { [weak self]() in
+                self?.isFetchingImage = false
+                self?.kind = .MissingImage
+                self?.triggerUpdate()
+            })        
+    }
+
+    func triggerUpdate(){
+        self.textContainer?.layoutManager?.invalidateLayoutForAttachment(self)
```

Comment:
We can discuss it separately from this PR.

I think the problem is that this is not necessarily a task for an "image provider" but it certainly is a task for our `TextStorage` to take care of.